packages/web/src/lib/api.ts

```diff
@@ -0,0 +1,54 @@
+import algosdk, { Algodv2, ALGORAND_MIN_TX_FEE, SuggestedParams } from "algosdk";
+
+import { ChainType, TxParams } from "../types";
+
+const mainNetClient = new Algodv2("", "https://algoexplorerapi.io", "");
+const testNetClient = new Algodv2("", "https://testnet.algoexplorerapi.io", "");
+
+export function clientForChain (chain: ChainType): algosdk.Algodv2 {
+  switch (chain) {
+    case ChainType.MainNet:
+      return mainNetClient;
+    case ChainType.TestNet:
+      return testNetClient;
+    default:
+      throw new Error(`Unknown chain type: ${chain as string}`);
+  }
+}
+
+/**
+ * Returns blockchain transaction suggested parameters (firstRound, lastRound, fee..)
+ * @param algocl an Algorand client, instance of Algodv2, used to communicate with a blockchain node.
+ */
+export async function getSuggestedParams (algocl: Algodv2): Promise<SuggestedParams> {
+  const params = await algocl.getTransactionParams().do();
+  // Private chains may have an issue with firstRound
+  if (params.firstRound === 0) {
+    throw new Error("Suggested params returned 0 as firstRound. Ensure that your node progresses.");
+    // params.firstRound = 1
+  }
+  return params;
+}
+
+/**
+ * Returns a union object of custom transaction params and suggested params.
+ * @param algocl an Algorand client, instance of Algodv2, used to communicate with a blockchain node.
+ * @param userParams a dict containing custom params defined by the user
+ * @param s suggested transaction params
+ */
+export async function mkTxParams (
+  algocl: Algodv2, userParams: TxParams, s?: SuggestedParams): Promise<SuggestedParams> {
+  if (s === undefined) { s = await getSuggestedParams(algocl); }
+
+  if (userParams.flatFee === undefined) {
+    if (userParams.totalFee !== undefined) s.flatFee = true;
+    else s.flatFee = false;
+  }
+  s.fee = userParams.totalFee ?? userParams.feePerByte ?? ALGORAND_MIN_TX_FEE;
```

Comment:
I think we shouldn't set `ALGORAND_MIN_TX_FEE` if both `userParams.totalFee` and `userParams.feePerByte` are unknown.

Reply:
Could you please explain the reason (not setting fee as minimum, if not passed by user) ?

OR do you mean the network should set fee on it's own in this case?

Reply:
Yes, the JS SDK will do it.

Reply:
ok

Reply:
ah, we can't set the fee as `undefined`, it must be number. [https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.ts#L476](https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.ts#L476)

Reply:
type [https://github.com/algorand/js-algorand-sdk/blob/develop/src/types/transactions/base.ts#L94](https://github.com/algorand/js-algorand-sdk/blob/develop/src/types/transactions/base.ts#L94)